Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnSelect.hs

```diff
@@ -110,11 +110,13 @@ onSelect OnSelectValidatedReq {..} = do
       let lowestFareQuote = selectLowestFareQuote quotes
       case lowestFareQuote of
         Just autoAssignQuote -> do
-          let dConfirmReq = SConfirm.DConfirmReq {personId = person.id, quoteId = autoAssignQuote.id, paymentMethodId = searchRequest.selectedPaymentMethodId}
-          dConfirmRes <- SConfirm.confirm dConfirmReq
-          becknInitReq <- ACL.buildInitReqV2 dConfirmRes
-          handle (errHandler dConfirmRes.booking) $
-            void . withShortRetry $ CallBPP.initV2 dConfirmRes.providerUrl becknInitReq
+          isLockAcquired <- SConfirm.tryInitTriggerLock person.id
```

Comment:
Why both locks are `tryInitTriggerLock`?

Reply:
And lock shouldn't be on person.id, it should be on search request id.

Reply:
> Why both locks are tryInitTriggerLock?
as we are trying to acquire lock thats why try is used, in case we don't get lock we need to throwError in autoAssign false case (for UI API [here](https://github.com/nammayatri/nammayatri/pull/6102/files#:~:text=isLockAcquired%20%3C%2D%20SConfirm.tryInitTriggerLock%20personId))
and wanted to keep same locking function for both.

> And lock shouldn't be on person.id, it should be on search request id.
fixed.